internal/services/postgres/postgresql_flexible_server_virtual_endpoint_resource_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package postgres_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2023-06-01-preview/virtualendpoints"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type PostgresqlFlexibleServerVirtualEndpointResource struct{}
+
+func TestAccPostgresqlFlexibleServerVirtualEndpoint_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_virtual_endpoint", "test")
+	r := PostgresqlFlexibleServerVirtualEndpointResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").Exists(),
+				check.That(data.ResourceName).Key("source_server_id").Exists(),
+				check.That(data.ResourceName).Key("replica_server_id").Exists(),
+				check.That(data.ResourceName).Key("type").Exists(),
```

Comment:
`data.ImportStep` takes care of comparing the config to state so we don't need to add checks like this - we usually only add checks for very specific cases or in data source tests
```suggestion
```